src/analyst.py

```diff
@@ -583,4 +583,131 @@ def summarize_stats(self) -> dict:
+
+        submit_time = []
+        for i, respondent in enumerate(student_list):
+            value_array[i, :] = respondent.student['student_sentiment']['values']
+            submit_time.append(respondent.metadata['submit_time'])
+
+        res = dict()
+
+        res['keys']   = keys
+        res['values'] = value_array
+        res['submit_time'] = np.array(submit_time)
+
+        # Summary statistics
+        res['mean']   = np.nanmean(value_array, axis=0)
+        res['stdev']  = np.nanstd(value_array, axis=0)
+
+        return res
+    
+    def summarize_student_backgrounds(self, respondents_list=None) -> dict:
+        """
+        Return summary of respondent's backgrounds
+        """
+
+        if respondents_list is None:
+            respondents_list = self.respondents_list
+
+        student_list = self.filter_respondents_on(is_student=True,is_completed_all_questions=True)
+
+        degree = {}
+        country = {}
+        state = {}
+        education = {}
+
+        for respondent in student_list:
+            utils.update_dict_counter(degree, respondent.student['degree'])
+            utils.update_dict_counter(country, respondent.student['country'])
+            utils.update_dict_counter(state, respondent.student['state'])
+            utils.update_dict_counter(education, respondent.student['education'])
+        
+        res = dict()
+        res['degree']           = degree
+        res['country']          = country
+        res['state']            = state
+        res['education']        = education
+
+        return res
+
+    def summarize_student_ideal(self, respondents_list=None) -> dict:
+        """
+        Summarize students' ideal career for those who completed the "Student" questions
+        """
+
+        if respondents_list is None:
+            respondents_list = self.respondents_list
+
+        student_list = self.filter_respondents_on(is_student=True,is_completed_all_questions=True)
+
+        # needs LLM for job title
+        # ideal_job_title_list = []
+
+        # building other summary for now
+        ideal_vc_counter = dict()
+        ideal_job_aspects_counter = dict()
+        ideal_salary_list = []
+
+        for respondent in student_list:
+            # utils.nanappend(ideal_job_title_list, respondent.student['ideal_job_title'])
+            utils.update_dict_counter(ideal_vc_counter, respondent.student['ideal_value_chain'])
+            utils.update_dict_counter(ideal_job_aspects_counter, respondent.student['ideal_job_aspects'])
+            utils.nanappend(ideal_salary_list, respondent.student['ideal_salary'])
+
+        res = dict()
+        # res['ideal_job_title_list'] = ideal_job_title_list
+        res['ideal_value_chain'] = ideal_vc_counter
+        res['ideal_job_aspects'] = ideal_job_aspects_counter
+        res['ideal_salary_list'] = np.array(ideal_salary_list)
+
+        # these numbers aren't very good... may need to manually pick out outliers
```

Comment:
I'm taking the approach of 'delayed filtering' for this analysis. The analyst uses these tools to look at the data and applies the filters as part of the final analysis. If we do too much filtering during the data processing, sometimes we lose track of how much was filtered. 

For the summary statistics, maybe we can remove mean and std since these are the two metrics affected by outliers. I support this but let's do this broadly across all the different sentiment functions.

Reply:
to be addressed in #19